File: src/main/java/com/ejo/game/render/Ground.java
package com.ejo.game.render;

import com.ejo.game.App;
import com.ejo.game.math.Vector;

import java.awt.*;
import java.util.ArrayList;
import java.util.Random;

public class Ground implements Drawable {

    private final int yPos;

    private final int rockCount;

    private final ArrayList<Vector> rockPosList = new ArrayList<>();

    public Ground(int yPos, int rockCount) {
        this.yPos = yPos;
        this.rockCount = rockCount;
        initRockPositions();
    }

    @Override
    public void draw(Graphics2D graphics, Vector pos) {
        new Rectangle(new Vector(App.WINDOW.getSize().getX(),100),new Color(76, 36, 1,255)).draw(graphics,new Vector(0,yPos));

        for (int i = 0; i < rockCount; i++) {
            Rectangle rock = new Rectangle(new Vector(6,6), new Color(58, 26, 0,255),Math.PI / 4);
            rock.draw(graphics, rockPosList.get(i));
        }
    }

    private void initRockPositions() {
        Random random = new Random();
        for (int i = 0; i < 30; i++) {
            rockPosList.add(new Vector(random.nextInt((int)App.WINDOW.getSize().getX()), random.nextInt((int)App.WINDOW.getSize().getY() - 500) + 500));
        }
    }

    public int getY() {
        return yPos;
    }
}

File: render/environment/Ground.py
import math
import random

from pygame import Color

from render.Rectangle import RotatedRectangle, Rectangle
from scene.Scene import Scene
from util.Vector import Vector

class Ground(Rectangle):

    def __init__(self, scene: Scene, y: int, rock_count: int):
        super().__init__(scene, Vector(0,scene.window.size.y - y),Vector(scene.window.size.x,y),Color(76, 36, 1, 255))
        self.rock_count = rock_count

        self.rocks: list[RotatedRectangle] = []

        #Generate Rocks
        rand = random.Random()
        for i in range(rock_count):
            size: Vector = Vector(5,5)
            pos: Vector = Vector(rand.randint(0,self.scene.window.size.x),rand.randint(int(self.pos.y + size.y),self.scene.window.size.y))
            rock: RotatedRectangle = RotatedRectangle(scene, pos, size, Color(58, 26, 0,255), math.tau/8)
            self.rocks.append(rock)

    def draw(self):
        super().draw()
        for rock in self.rocks:
            rock.draw()
